crypto/err/err.c

```diff
@@ -812,3 +812,26 @@ int ERR_pop_to_mark(void)
     es->err_flags[es->top] &= ~ERR_FLAG_MARK;
     return 1;
 }
+
+int ERR_clear_last_mark(void)
+{
+    ERR_STATE *es;
+    int top;
+
+    es = ERR_get_state();
+    if (es == NULL)
+        return 0;
+
+    top = es->top;
+    while (es->bottom != top
+           && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0) {
+        top -= 1;
+        if (top == -1)
+            top = ERR_NUM_ERRORS - 1;
```

Comment:
On a *side note*. It appears to me that it was copied from elsewhere, but I find it hard to recognize circular buffer pattern here (and elsewhere). Maybe line of commentary would be appropriate? Or maybe `top = top ? top - 1 : ERR_NUM_ERRORS - 1`?